socketshark/subscription.py

```diff
@@ -130,8 +167,37 @@ def should_deliver_message(self, data):
                                          reason='order')
             return False
 
+        if not self._should_deliver_message_throttle(data):
+            self.session.trace_log.debug('message filtered', data=data,
+                                         reason='throttle')
+            return False
+
         return True
 
+    async def _schedule_throttled_message(self, delay, throttle_key):
+        try:
+            await asyncio.sleep(delay)
+            await self._send_throttled_message(throttle_key)
```

Comment:
I feel like this `await` could cause a race condition where we have woken up and started sending a throttled message (but haven't finished) and simultaneously start deciding if we should throttle a message in `_should_deliver_message_throttle`.  Not sure if that could lead to unpredictable states in `self.throttle_state` causing extra or missed messages.